From ad987063207b77763e541f57535ea6b681e4ef3b Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 18 Oct 2021 11:12:42 +0100
Subject: [PATCH] ARM: dts: bcm2835_audio missing firmware reference

The firmware driver has been changed to count its clients. An earlier
commit removed the downstream patch permitting the hacky technique of
passing NULL to rpi_firmware_get to pick up the last instace, but
failed to add the necessary "firmware" property to the bcm2835_audio
node. Correct that omission.

See: https://github.com/raspberrypi/linux/issues/4634
     https://github.com/raspberrypi/linux/issues/4635

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm270x-rpi.dtsi | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm/boot/dts/bcm270x-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm270x-rpi.dtsi
@@ -148,6 +148,7 @@
 	/* Onboard audio */
 	audio: bcm2835_audio {
 		compatible = "brcm,bcm2835-audio";
+		brcm,firmware = <&firmware>;
 		brcm,pwm-channels = <8>;
 		status = "disabled";
 	};
